File: sdk/predict_client.py
import requests
import json

class AgriDemandClient:
    """
    Client for the Agri-Demand Optimizer (Kenya Fresh Produce).
    Predicts shopping baskets to reduce food waste and stabilize farmer income.
    """
    def __init__(self, api_url="https://okechobonyo-farm-to-feed-elite.hf.space/predict_demand"):
        self.api_url = api_url

    def get_forecast(self, customer_category: str, sku_unit: str, lag_qty: float, freq_4w: int, peak_qty: float):
        """
        Request 7-day and 14-day demand projections for a specific SKU.
        """
        payload = {
            "customer_category": customer_category,
            "unit_name": sku_unit,
            "grade_name": "GRADE_01", # Default to standard pilot grade
            "qty_lag_1w": lag_qty,
            "freq_4w": freq_4w,
            "freq_8w": freq_4w * 2,  # Simulated estimation
            "freq_12w": freq_4w * 3, # Simulated estimation
            "pair_max_qty": peak_qty,
            "pair_avg_qty": (lag_qty + peak_qty) / 2
        }

        try:
            response = requests.post(self.api_url, json=payload, timeout=10)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            return {"status": "error", "message": str(e)}

# --- Usage Example ---
# client = AgriDemandClient()
# report = client.get_forecast("CUST_CAT_001", "UNIT_004", 12.5, 3, 25.0)
# print(f"7-Day Volume: {report['predictions']['week_1']['predicted_quantity']} units")

File: sdk/predict_client.js
/**
 * Client for the Agri-Demand Optimizer (Kenya Fresh Produce).
 * Facilitates predictive procurement for commercial kitchens.
 */
class AgriDemandClient {
    constructor(apiUrl = "https://okechobonyo-farm-to-feed-elite.hf.space/predict_demand") {
        this.apiUrl = apiUrl;
    }

    async getForecast(customerCategory, skuUnit, lagQty, freq4w, peakQty) {
        const payload = {
            customer_category: customerCategory,
            unit_name: skuUnit,
            grade_name: "GRADE_01",
            qty_lag_1w: lagQty,
            freq_4w: freq4w,
            freq_8w: freq4w * 2,
            freq_12w: freq4w * 3,
            pair_max_qty: peakQty,
            pair_avg_qty: (lagQty + peakQty) / 2
        };

        try {
            const response = await fetch(this.apiUrl, {
                method: "POST",
                headers: { "Content-Type": "application/json" },
                body: JSON.stringify(payload)
            });

            if (!response.ok) throw new Error(`HTTP Error: ${response.status}`);
            return await response.json();
        } catch (error) {
            return { status: "error", message: error.message };
        }
    }
}

// --- Usage Example ---
// const client = new AgriDemandClient();
// const report = await client.getForecast("CUST_CAT_001", "UNIT_004", 12.5, 3, 25.0);
// console.log(`14-Day Outlook: ${report.predictions.week_2.predicted_quantity} units`);
